### ISSUE TITLE
dbt templater cannot handle "raw" jinja tags

### ISSUE THREAD

**Issue by cnolanminich** [sqlfluff/sqlfluff#681]
## Expected Behaviour

`sqlfluff lint` on SQL files with `{% raw %}` Jinja tags should be able to be linted and fixed.

## Observed Behaviour

> KeyError: 'raw_end' 

## Steps to Reproduce

-- table_ref.sql
```sql
select 1
from foo
```

-- test_raw.sql
```sql
select
    table_ref.field1,
    {% raw %}
    table_ref.field2,
    table_ref.field3
    {% endraw %}
from
    {{ref('table_ref')}} as table_ref
```

## Version
python
> Python 3.8.2
dbt --version
> installed version: 0.18.1
> latest version: 0.18.1
> 
> Up to date!

sqlfluff -- version
> sqlfluff, version 0.4.0a3

## Configuration
```
[sqlfluff]
templater = dbt
exclude_rules = L003,L013,L022,L027,L009,L031

[sqlfluff:templater:jinja:context]
target = 'dev'

[sqlfluff:indentation]
indented_joins = True

[sqlfluff:rules]
max_line_length = 115

[sqlfluff:rules:L010]
capitalisation_policy = lower
```
.sqlfluffignore
```
target/
dbt_modules/
macros/
```

**Comment by alanmcruickshank** [comment]
I think I know what's causing this, and there are a few aspects.

1. The slicing code needs to know how to handle "raw" and "endraw". This involves the slicing code in `sqlfluff.core.templaters.jinja.py`. However in reviewing that now I _think_ this patch of code is actually fine.
2. We need to make sure that anything _between_ the `raw` and `endraw` tags is not parsed as templated code. This one is going to be slightly trickier.

Could you make a PR with a failing test case on this one - I realize I was pretty confident in the fix for this one earlier but now I'm less certain.

**Comment by pwildenhain** [comment]
@alanmcruickshank See https://github.com/sqlfluff/sqlfluff/pull/737 for a failing test case. If you see what needs to be done, and would like someone else to work on it, feel free to tag me 👍🏻
